src/api/shopper-insights/validation.js

```diff
@@ -0,0 +1,198 @@
+  return merchantPayload.hasOwnProperty("purchaseUnits");
+};
+
+const hasCountryCode = (merchantPayload: MerchantPayloadData): boolean => {
+  return merchantPayload.hasOwnProperty("countryCode");
+};
+
+const isNonProdEnvironment = getEnv() !== "production";
+const buyerCountryForLowerEnvironments = getBuyerCountry() || "US";
+
+export function validateMerchantPayload(merchantPayload: MerchantPayloadData) {
+  if (typeof merchantPayload !== "object") {
+    throw new ValidationError(
+      `Expected shopper information to be passed as an object`
+    );
+  }
+
+  const hasTopLevelCustomer = Object.keys(merchantPayload).includes("customer");
+  if (Object.keys(merchantPayload).length === 0 || !hasTopLevelCustomer) {
+    throw new ValidationError(
+      `Expected shopper information to be passed into customer object`
+    );
+  }
+
+  const hasEmailOrPhoneNumber =
+    hasEmail(merchantPayload) || hasPhoneNumber(merchantPayload);
+  if (!hasEmailOrPhoneNumber) {
+    throw new ValidationError(
+      `Expected shopper information to include an email or phone number`
+    );
+  }
+
+  if (
+    hasPurchaseUnits(merchantPayload) &&
+    !merchantPayload?.purchaseUnits?.[0]?.amount?.currencyCode
+  ) {
+    throw new ValidationError(
+      `Expected purchase units to include a currency code in the amount object`
+    );
+  }
+
+  const allowedPayload = ["customer", "purchaseUnits"];
+  if (isNonProdEnvironment) {
+    allowedPayload.push("countryCode", "channel");
+  }
+
+  Object.keys(merchantPayload).forEach((merchantPayloadKey) => {
+    if (allowedPayload.indexOf(merchantPayloadKey) === -1) {
+      throw new ValidationError(
+        `Unexpected shopper information passed: ${merchantPayloadKey} is not supported`
+      );
+    }
+  });
+}
+
+type getRecommendedPaymentMethodsRequestPayload = {|
+  customer: {|
+    email?: string,
+    phone?: {|
+      country_code: string,
+      national_number: string,
+    |},
+  |},
+  purchase_units: $ReadOnlyArray<{|
+    amount: {|
+      currency_code: string,
+    |},
+  |}>,
+  country_code?: string,
+  include_account_details: boolean,
+|};
+
+export function createRequestPayload(
+  merchantPayload: MerchantPayloadData
+): getRecommendedPaymentMethodsRequestPayload {
+  return {
+    customer: {
+      ...(hasEmail(merchantPayload) && {
+        email: merchantPayload.customer?.email,
+      }),
+      // $FlowIssue
+      ...(hasPhoneNumber(merchantPayload) && {
+        phone: {
+          country_code: merchantPayload?.customer?.phone?.countryCode,
+          national_number: merchantPayload?.customer?.phone?.nationalNumber,
+        },
+      }),
+    },
+    purchase_units: [
+      {
+        amount: {
+          currency_code: hasPurchaseUnits(merchantPayload)
+            ? // $FlowFixMe
+              merchantPayload?.purchaseUnits?.[0]?.amount?.currencyCode
+            : getCurrency(),
+        },
+      },
+    ],
```

Comment:
we can do a new build tomorrow but I think we should do:

```
purchase_units: hasPurchaseUnits(merchantPayload)
  ? merchantPayload.purchaseUnits
  : [...our purchase unit with the currency]
```

that way if for whatever reason they want to pass their whole purchase unit to the api they can and we'll stay out of the way